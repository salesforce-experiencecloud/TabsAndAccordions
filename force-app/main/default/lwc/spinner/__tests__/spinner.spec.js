/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import cElement from 'c/spinner';

const alternativeText = 'required alternative text';

function createComponent(props = {}) {
    const element = createElement('c-spinner', {
        is: cElement
    });

    Object.assign(element, props);
    document.body.appendChild(element);
    return element;
}

describe('c-spinner', () => {
    afterEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('default', () => {
        const element = createComponent({
            alternativeText
        });

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    ['small', 'medium', 'large'].forEach((size) => {
        it(`size=${size}`, () => {
            const element = createComponent({
                size,
                alternativeText
            });

            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });
    });

    ['base', 'brand', 'inverse'].forEach((variant) => {
        it(`variant=${variant}`, () => {
            const element = createComponent({
                variant,
                alternativeText
            });

            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });
    });
});
