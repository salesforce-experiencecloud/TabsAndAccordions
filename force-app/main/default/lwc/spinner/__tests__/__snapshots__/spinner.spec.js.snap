// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-spinner default 1`] = `
<c-spinner
  class="slds-spinner_container"
>
  #shadow-root(open)
    <div
      class="slds-spinner slds-spinner_medium"
      role="status"
    >
      <span
        class="slds-assistive-text"
      >
        required alternative text
      </span>
      <div
        class="slds-spinner__dot-a"
      />
      <div
        class="slds-spinner__dot-b"
      />
    </div>
</c-spinner>
`;

exports[`c-spinner size=large 1`] = `
<c-spinner
  class="slds-spinner_container"
>
  #shadow-root(open)
    <div
      class="slds-spinner slds-spinner_large"
      role="status"
    >
      <span
        class="slds-assistive-text"
      >
        required alternative text
      </span>
      <div
        class="slds-spinner__dot-a"
      />
      <div
        class="slds-spinner__dot-b"
      />
    </div>
</c-spinner>
`;

exports[`c-spinner size=medium 1`] = `
<c-spinner
  class="slds-spinner_container"
>
  #shadow-root(open)
    <div
      class="slds-spinner slds-spinner_medium"
      role="status"
    >
      <span
        class="slds-assistive-text"
      >
        required alternative text
      </span>
      <div
        class="slds-spinner__dot-a"
      />
      <div
        class="slds-spinner__dot-b"
      />
    </div>
</c-spinner>
`;

exports[`c-spinner size=small 1`] = `
<c-spinner
  class="slds-spinner_container"
>
  #shadow-root(open)
    <div
      class="slds-spinner slds-spinner_small"
      role="status"
    >
      <span
        class="slds-assistive-text"
      >
        required alternative text
      </span>
      <div
        class="slds-spinner__dot-a"
      />
      <div
        class="slds-spinner__dot-b"
      />
    </div>
</c-spinner>
`;

exports[`c-spinner variant=base 1`] = `
<c-spinner
  class="slds-spinner_container"
>
  #shadow-root(open)
    <div
      class="slds-spinner slds-spinner_medium"
      role="status"
    >
      <span
        class="slds-assistive-text"
      >
        required alternative text
      </span>
      <div
        class="slds-spinner__dot-a"
      />
      <div
        class="slds-spinner__dot-b"
      />
    </div>
</c-spinner>
`;

exports[`c-spinner variant=brand 1`] = `
<c-spinner
  class="slds-spinner_container"
>
  #shadow-root(open)
    <div
      class="slds-spinner slds-spinner_brand slds-spinner_medium"
      role="status"
    >
      <span
        class="slds-assistive-text"
      >
        required alternative text
      </span>
      <div
        class="slds-spinner__dot-a"
      />
      <div
        class="slds-spinner__dot-b"
      />
    </div>
</c-spinner>
`;

exports[`c-spinner variant=inverse 1`] = `
<c-spinner
  class="slds-spinner_container"
>
  #shadow-root(open)
    <div
      class="slds-spinner slds-spinner_inverse slds-spinner_medium"
      role="status"
    >
      <span
        class="slds-assistive-text"
      >
        required alternative text
      </span>
      <div
        class="slds-spinner__dot-a"
      />
      <div
        class="slds-spinner__dot-b"
      />
    </div>
</c-spinner>
`;
